flexget/plugins/sites/torrentleech.py

```diff
@@ -100,13 +97,15 @@ def search(self, task, entry, config=None):
         """
         Search for name from torrentleech.
         """
+        task.requests.headers.update({'User-Agent': 'curl/7.54.0'})
```

Comment:
This changes the user agent for all plugins in the same task run. Not ideal.

Reply:
Unfortunately my initial tests where correct. The User-Agent has to be set globally otherwise the task will fail when using the download output plugin ``flexget.plugins.output.download.download_entry``. When it tries to get the file it fails if the User-Agent is not set to ``curl/7.54.0`` or something else supported.

Setting it globally affects other plugins so that is not good. I suppose the headers could be passed through the entry ``entry['headers']= {'User-Agent': 'curl/7.54.0'}`` much like auth is handles in the ``flexget.plugins.output.download.download_entry`` method.

Reply:
Another approach would be to create a custom download_auth like so:

``` python
class TorrentleechHeader(AuthBase):

    def __call__(self, request):
        request.headers['User-Agent'] = ''curl/7.54.0"
        return request
```
@cvium @liiight What do you think is best?

Reply:
No opinion